crates/stages/src/stages/tx_lookup.rs

```diff
@@ -92,43 +96,60 @@ impl<DB: Database> Stage<DB> for TransactionLookupStage {
             return Ok(ExecOutput::done(input.checkpoint()))
         }
 
-        let (tx_range, block_range, is_final_range) =
-            input.next_block_range_with_transaction_threshold(provider, self.commit_threshold)?;
-        let end_block = *block_range.end();
+        // 500MB temporary files
+        let mut hash_collector: Collector<TxHash, TxNumber> =
+            Collector::new(Arc::new(TempDir::new()?), 500 * (1024 * 1024));
 
-        debug!(target: "sync::stages::transaction_lookup", ?tx_range, "Updating transaction lookup");
+        loop {
```

Comment:
does this mean we're not respecting the thresholds anymore, and do all work in one huge commit? what will happen if I shut down the node mid-way through the stage, am I going to lose the progress?

Reply:
yes, everything needs to happen again. same as headers etl stage.

thresholds are still important for: hashing chunks of transactions and/or interaction with `node stage drop` `node stage run`

Reply:
shouldn't be called `commit_threshold` then?